lib/dry/validation/result.rb

```diff
@@ -1,83 +1,39 @@
-require 'dry/equalizer'
+require 'dry/validation/constants'
 
 module Dry
   module Validation
     class Result
-      include Dry::Equalizer(:output, :errors)
-      include Enumerable
-
-      attr_reader :output
-      attr_reader :results
-      attr_reader :message_compiler
-      attr_reader :path
-
-      alias_method :to_hash, :output
-      alias_method :to_h, :output # for MRI 2.0, remove it when drop support
-
-      def initialize(output, results, message_compiler, path)
-        @output = output
-        @results = results
-        @message_compiler = message_compiler
-        @path = path
-      end
-
-      def each(&block)
-        output.each(&block)
-      end
-
-      def [](name)
-        output.fetch(name)
-      end
-
-      def success?
-        results.empty?
+      def self.new(params, errors = EMPTY_HASH.dup)
+        result = super
+        yield(result) if block_given?
+        result.freeze
       end
 
-      def failure?
-        !success?
-      end
+      attr_reader :params
 
-      def messages(options = EMPTY_HASH)
-        message_set(options).dump
-      end
+      attr_reader :errors
 
-      def errors(options = EMPTY_HASH)
-        message_set(options.merge(hints: false)).dump
+      def initialize(params, errors)
+        @params = params
+        @errors = errors.update(params.errors)
```

Comment:
this mutates input value, is it safe?

Reply:
@flash-gordon yeah it's being built via `.new` and then it's frozen. See how it's used in `Contract#call`